processor/src/test/java/org/mapstruct/ap/test/nestedbeans/RecursionTest.java

```diff
@@ -0,0 +1,88 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.nestedbeans;
+
+import java.util.Collections;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mapstruct.ap.test.nestedbeans.recursive.RecursionMapper;
+import org.mapstruct.ap.test.nestedbeans.recursive.TreeRecursionMapper;
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+
+@RunWith(AnnotationProcessorTestRunner.class)
+public class RecursionTest {
+
+    @WithClasses({
+        RecursionMapper.class
+    })
+    @Test
+    @IssueKey("1103")
+    public void testRecursiveAutoMap() {
+        RecursionMapper.Root root = RecursionMapper.INSTANCE.mapRoot(
+            new RecursionMapper.RootDto(
+                new RecursionMapper.ChildDto( "Sub Root", new RecursionMapper.ChildDto(
+                    "Sub child", null
+                ) )
+            ) );
+
+        Assert.assertEquals(
```

Comment:
I will do how you ask, though as I think of it - the best solution here would be map dto to entity and back (we assume they have the same number of properties) and then compare them by the equals method. Something like
`RootDto rootDto = createTestRootDto();
assertEquals(rootDto, mapper.mapEntityToDto( mapper.mapDtoToEntity( rootDto ) );`
This way there is no need to write own equality methods (possibly error prone) and no need to create an additional object of different type, but same parameters.